File: games/planetization/match-generator/core/src/com/planet_lia/match_generator/game/api/PlanetData.java
package com.planet_lia.match_generator.game.api;

import com.planet_lia.match_generator.game.entities.Owner;

import java.util.ArrayList;

public class PlanetData {

    public int id;
    public Owner owner;
    public float x, y;
    public ArrayList<Integer> idsOfUnitsOnPlanet;
    public float resources;
    public boolean canSpawnNewUnit;

    public PlanetData(int id,
                      Owner owner,
                      float x,
                      float y,
                      ArrayList<Integer> idsOfUnitsOnPlanet,
                      float resources,
                      boolean canSpawnNewUnit) {
        this.id = id;
        this.owner = owner;
        this.x = x;
        this.y = y;
        this.idsOfUnitsOnPlanet = idsOfUnitsOnPlanet;
        this.resources = resources;
        this.canSpawnNewUnit = canSpawnNewUnit;
    }
}

File: games/planetization/bots/kotlin/src/core/api/PlanetData.kt
package core.api

import java.util.ArrayList

class PlanetData(
    var id: Int,
    var owner: Owner,
    var x: Float,
    var y: Float,
    var idsOfUnitsOnPlanet: ArrayList<Int>,
    var resources: Float,
    var canSpawnNewUnit: Boolean
)

enum class Owner {
    NONE, RED, GREEN
}
